src/Illuminate/Auth/Events/Logout.php

```diff
@@ -16,8 +17,8 @@ class Logout
      * @return void
      */
     public function __construct(
-        public $guard,
-        public $user,
+        public string $guard,
+        public Authenticatable $user,
```

Comment:
This is potentially `null`.

- https://github.com/laravel/pulse/issues/364
- https://github.com/laravel/pulse/issues/324
- https://github.com/laravel/pulse/pull/366
- https://github.com/laravel/fortify/pull/536

Reply:
yah, i need to dig into this PR a little more. this is where it gets tricky, because do we change what the Event accepts? or do we force the calling code to actually follow what the event wants.

Reply:
I was chatting to Taylor about this and suggested we go deeper on static analysis insights before we make refactors like this. Hopefully that would help expose places where the dockblocks are lying.

We don't want to be unintentionally introducing breaking changes.

Reply:
yah, anything we can do to move us towards more strict typing I'm down for that.

I'd think I'd be on the side of forcing the calling code to adjust, for the most part. 

glad you guys are talking about it internally. would love to help how I can.